crates/c-api/include/wasmtime/store.h

```diff
@@ -243,6 +253,45 @@ WASM_API_EXTERN void wasmtime_context_set_epoch_deadline(wasmtime_context_t *con
  */
 WASM_API_EXTERN void wasmtime_store_epoch_deadline_callback(wasmtime_store_t *store, wasmtime_error_t* (*func)(wasmtime_context_t*, void*, uint64_t*), void *data);
 
+/**
+ * \brief Insert a file descriptor into the underlying WASI context. 
+ * 
+ * This is used to dynamically bind a file descriptor from the host to a file 
+ * descriptor in an already-built WASI context to make it available to WASI APIs.
+ * And it requires that the WASI context has been configured correctly prior to
+ * calling this function.
+ * 
+ * The `guest_fd` argument is the number of the file descriptor by which it will
+ * be known in WASM and the `host_fd` is the number of the file descriptor on
+ * the host and the `access_mode` is the access mode of the file descriptor. 
+ * The enumeration of access modes aligns with FileAccessMode in the WASI spec 
+ * and as follows: READ = 0b1, WRITE = 0b10.
+ */
+#if defined(UNIX)
+WASM_API_EXTERN void wasmtime_context_insert_file(wasmtime_context_t *context, uint32_t guest_fd, int host_fd, uint32_t access_mode);
+#elif defined(WINDOWS)
+WASM_API_EXTERN void wasmtime_context_insert_file(wasmtime_context_t *context, uint32_t guest_fd, void *host_handle, uint32_t access_mode);
+#endif
```

Comment:
Could this perhaps do 

```c
#ifdef WASMTIME_UNIX
typedef int wasmtime_host_file_t;
#else   
typedef HANDLE wasmtime_host_file_t;
#endif 

```

to avoid defining the function signature twice?

Additionally this should use `HANDLE` on Windows, not `void*`, as fixing that in Go seems like an orthogonal concern?

Reply:
I'm not really getting it by what do you mean here:

```
#ifdef WASMTIME_UNIX
typedef int wasmtime_host_file_t;
#else   
typedef HANDLE wasmtime_host_file_t;
#endif 
```

Doesn't this just further complicates the interface by adding more layers of indirectness? 

> Go seems like an orthogonal concern

To be super honest with you, for me, my major concern is if it works in Go. So I would just save the effort. If it won't work for you, someone better at this could fix it...